tests/lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,1246 @@
+        }, {
+            code: "(a || (a = 0)) || b",
+            output: "(a ||= 0) || b",
+            errors: [{ messageId: "logical", type: "LogicalExpression", data: { operator: "||" } }]
+        }, {
+            code: "a || (b || (b = 0))",
+            output: "a || (b ||= 0)",
+            errors: [{ messageId: "logical", type: "LogicalExpression", data: { operator: "||" } }]
+        }, {
+            code: "a = b || (b = c)",
+            output: "a = b ||= c",
+            errors: [{ messageId: "logical", type: "LogicalExpression", data: { operator: "||" } }]
+        }, {
+            code: "a || (a = 0) ? b : c",
+            output: "(a ||= 0) ? b : c",
+            errors: [{ messageId: "logical", type: "LogicalExpression", data: { operator: "||" } }]
+        }, {
+            code: "fn(a || (a = 0))",
+            output: "fn(a ||= 0)",
+            errors: [{ messageId: "logical", type: "LogicalExpression", data: { operator: "||" } }]
+        },
+
+        // If
+        {
+            code: "if (a) a = b",
+            output: "a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (Boolean(a)) a = b",
+            output: "a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (!!a) a = b",
+            output: "a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (!a) a = b",
+            output: "a ||= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (!Boolean(a)) a = b",
+            output: "a ||= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a == undefined) a = b",
+            output: "a ??= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a == null) a = b",
+            output: "a ??= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a === null || a === undefined) a = b",
+            output: null,
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{
+                messageId: "if",
+                type: "IfStatement",
+                suggestions: [{ messageId: "convertIf", output: "a ??= b" }]
+            }]
+        }, {
```

Comment:
This can be autofixed instead of providing a suggestion.

In this pattern, we don't want to autofix member expressions:

```js
"if (a.b === null || a.b === undefined) a.b = c"
```

because if `a.b` happens to be a getter, in the original code it can be triggered twice, while in the fixed version (`a.b ??= c`) it will always be triggered only once.